server/src/com/cloud/resource/ResourceManagerImpl.java

```diff
@@ -271,6 +272,8 @@ public void setDiscoverers(final List<? extends Discoverer> discoverers) {
 
     private SearchBuilder<HostGpuGroupsVO> _gpuAvailability;
 
+    private Map<Long,Integer> retryHostMaintenance = new HashMap<>();
```

Comment:
This may be called by concurrent threads of execution, @nvazquez can you use a concurrent hashmap and/or instead of integer perhaps one of the atomic ints?

Reply:
Done, thanks